File: apiEZPZ/src/main/java/com/example/apiezpz/search/entity/DailyRank.java
package com.example.apiezpz.search.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "daily_rank", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"category", "date"})
})
public class DailyRank {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String category;

    @Column(nullable = false)
    private Long searchCount = 0L;

    @Column(nullable = false)
    private LocalDate date;

    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: apiEZPZ/src/main/kotlin/com/example/apiezpz/search/entity/DailyRank.kt
package com.example.apiezpz.search.entity

import jakarta.persistence.*
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
@Table(
    name = "daily_rank", 
    uniqueConstraints = [UniqueConstraint(columnNames = ["category", "date"])]
)
data class DailyRank(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(nullable = false)
    var category: String = "",

    @Column(nullable = false)
    var searchCount: Long = 0L,

    @Column(nullable = false)
    var date: LocalDate = LocalDate.now(),

    var createdAt: LocalDateTime? = null
) {
    @PrePersist
    protected fun onCreate() {
        this.createdAt = LocalDateTime.now()
    }
} 